implementation/src/main/java/io/smallrye/metrics/setup/config/MetricsConfigurationManager.java

```diff
@@ -0,0 +1,320 @@
+                        new Object[] { MP_DEFAULT_BUCKET_PROP, metricName, retVal });
+            }
+            return retVal;
+        }
+
+        return null;
+
+    }
+
+    /**
+     * Returns the matching configuration object if it exists, null otherwise
+     * 
+     * @param metricName the metric name to check configuration against
+     * @return the matching configuration object if it exists, null otherwise
+     */
+    public synchronized HistogramBucketMaxConfiguration getDefaultHistogramMaxBucketConfiguration(String metricName) {
+
+        String appName = getApplicationName();
+
+        Collection<HistogramBucketMaxConfiguration> computedValues = defaultHistogramBucketMaxConfig
+                .computeIfAbsent(appName, f -> {
+                    Optional<String> input = ConfigProvider.getConfig().getOptionalValue(MP_HISTOGRAM_MAX_CONFIG,
+                            String.class);
+
+                    return (input.isPresent()) ? HistogramBucketMaxConfiguration.parse(input.get()) : null;
+
+                });
+
+        if (computedValues != null && computedValues.size() != 0) {
+            HistogramBucketMaxConfiguration retVal = HistogramBucketMaxConfiguration.matches(computedValues,
+                    metricName);
+            if (LOGGER.isLoggable(Level.FINEST)) {
+                LOGGER.logp(Level.FINEST, CLASS_NAME, null,
+                        "Returning \"{0}\" configuration for metric:\"{0}\" with values: {1} ",
+                        new Object[] { MP_HISTOGRAM_MAX_CONFIG, metricName, retVal });
+            }
+            return retVal;
+        }
+        return null;
+
+    }
+
+    /**
+     * Returns the matching configuration object if it exists, null otherwise
+     * 
+     * @param metricName the metric name to check configuration against
+     * @return the matching configuration object if it exists, null otherwise
+     */
+    public synchronized HistogramBucketMinConfiguration getDefaultHistogramMinBucketConfiguration(String metricName) {
+
+        String appName = getApplicationName();
+
+        Collection<HistogramBucketMinConfiguration> computedValues = defaultHistogramBucketMinConfig
+                .computeIfAbsent(appName, f -> {
+                    Optional<String> input = ConfigProvider.getConfig().getOptionalValue(MP_HISTOGRAM_MIN_CONFIG,
+                            String.class);
+
+                    return (input.isPresent()) ? HistogramBucketMinConfiguration.parse(input.get()) : null;
+
+                });
+
+        if (computedValues != null && computedValues.size() != 0) {
+            HistogramBucketMinConfiguration retVal = HistogramBucketMinConfiguration.matches(computedValues,
+                    metricName);
+            if (LOGGER.isLoggable(Level.FINEST)) {
+                LOGGER.logp(Level.FINEST, CLASS_NAME, null,
+                        "Returning \"{0}\" configuration for metric:\"{0}\" with values: {1} ",
```

Comment:
Should the parameter be {0}, {1}, and {2}?

Reply:
fixed